tests/secrets_manager_test.py

```diff
@@ -49,6 +49,48 @@ def test_get_db_credentials(self, mock_boto3, mock_requests, mock_logger):
         assert actual_secret['username'] == 'db_user'
         assert actual_secret['password'] == 'db_pass'
 
+    @mock.patch('hanadb_exporter.secrets_manager.LOGGER')
+    @mock.patch('hanadb_exporter.secrets_manager.requests')
+    @mock.patch('hanadb_exporter.secrets_manager.boto3.session')
+    def test_get_db_credentials_imdsv2(self, mock_boto3, mock_requests, mock_logger):
+        mock_ec2_unauthorized = mock.Mock()
+        mock_ec2_unauthorized.status_code = 401
+
+        mock_ec2_response = mock.Mock()
+        mock_ec2_response.json.return_value = json.loads('{"region":"test_region_imdsv2"}')
+
+        mock_requests.get.return_value = mock_ec2_response
```

Comment:
Thanks, when I follow your suggestion and remove
```
        mock_requests.get.return_value = mock_ec2_response
        mock_requests.get.return_value.status_code = 401
```

Then the assert 
```
        mock_session.client.assert_called_once_with(service_name='secretsmanager', region_name='test_region_imdsv2')
```

fails with
```
E           AssertionError: expected call not found.
E           Expected: client(service_name='secretsmanager', region_name='test_region_imdsv2')
E           Actual: client(service_name='secretsmanager', region_name=<MagicMock name='requests.get().json().__getitem__()' id='139931020782800'>)
```

In the debugger I can see that in this case the if statement `if ec2_info_response.status_code == 401` (Line 39, hanadb_exporter/secrets_manager.py), which is the relevant part for IMDSv2, get skipped because the status code isn't set.

Based on what I can see, the return_value seem still required even when using side_effects.

Reply:
This doesn't make much sense, the code must have something else wrong.
The side_effects can be used to give a list of values you want to return for this mock in each call.
Have a look here:
https://thedmitry.pw/blog/2020/12/mocks-side-effect/

Having both `return_value` and `side_effects` is something we shouldn't do

Reply:
Damn, the issue was that finale `s` in `side_effect`...